distbelief/optim/downpour_sgd.py

```diff
@@ -70,12 +69,11 @@ def step(self, closure=None):
         if self.idx % self.freq == 0:
             send_message(MessageCode.ParameterRequest, self.accumulated_gradients) # dummy val
 
-        # NB: loss.backward() should be called before this, so that we are sure that the .grad in the model params
-        # can be accessed. This should be always satisfied since we call loss.backward() right before step()
+        #get the lr
+        lr = self.param_groups[0]['lr']
         # keep track of accumulated gradients so that we can send 
         gradients = ravel_model_params(self.model, grads=True)
-        # this does self.accumulated_gradients+=-self.lr * gradients
-        self.accumulated_gradients.add_(gradients)
+        self.accumulated_gradients.add_(-lr, gradients)
```

Comment:
The server also does `tensor.add_(-lr, gradients)`, so having it here will cause the gradients to be scaled by the learning rate twice. I think we should have `self.accumulated_gradients.add_(gradients)` here so that the server can do whatever it wants with the gradients (initially just scaling by the LR, but later we can do different sorts of optimizers on the client/servers.

Reply:
It looks like the current `server.py` still scales by the LR: https://github.com/ucla-labx/distbelief/blob/db54a1ae9937d2f495f28f65e98abacdec82632d/distbelief/server.py#L42. Agreed with your approach, but if we only add raw gradients to the server then the best we can do on the server is probably like `SGD`

Reply:
Sounds good! Will this mean that the server and clients will always be running the same optimizer?